### PR TITLE
App Submission: Bitfeed

### PR DESCRIPTION
**Pull request by mononaut**
# App Submission

### App name
Bitfeed

### Version
2.1.2

### One line description of the app
A live visualization of your node's mempool.

### Summary of the app
A self-hosted version of Bitfeed - the open source mempool & block visualizer available at https://bits.monospace.live. Watch as new transactions drop into your node's mempool, before being packaged into newly mined blocks. Monitor Bitcoin network activity, explore the composition of the latest block, or simply enjoy a soothing Bitcoin screensaver.

### Developer name
Mononaut

### Developer website
https://monospace.live

### Source code link
https://github.com/bitfeed-project/bitfeed

### Support link
https://github.com/bitfeed-project/bitfeed/issues

### Requires
- [x] Bitcoin Core
- [ ] Electrum server
- [ ] LND

### 256x256 SVG icon
image

### Gallery images

umbrel local_8314_1
umbrel local_8314_2
umbrel local_8314_3

### I have tested my app on:
- [ ] [Umbrel dev environment](https://github.com/getumbrel/umbrel-dev)
- [x] [Umbrel OS on a Raspberry Pi 4](https://github.com/getumbrel/umbrel-os)
- [ ] [Custom Umbrel install on Linux](https://github.com/getumbrel/umbrel#-installation)

### CODE REVIEW

**Comment by lukechilds** [comment]
Thanks for this @mononaut, looks awesome! Will review as asap.

File: apps/bitfeed/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    environment:
+      PORT: "$APP_BITFEED_API_PORT"
+      BITCOIN_HOST: "$BITCOIN_IP"
+      BITCOIN_ZMQ_RAWTX_PORT: "$BITCOIN_ZMQ_RAWTX_PORT"
+      BITCOIN_ZMQ_RAWBLOCK_PORT: "$BITCOIN_ZMQ_RAWBLOCK_PORT"
+      BITCOIN_RPC_PORT: "$BITCOIN_RPC_PORT"
+      BITCOIN_RPC_USER: "$BITCOIN_RPC_USER"
+      BITCOIN_RPC_PASS: "$BITCOIN_RPC_PASS"
+    ports:
+      - "$APP_BITFEED_API_PORT:$APP_BITFEED_API_PORT"
```
Comment (lukechilds) [inline apps/bitfeed/docker-compose.yml:34]: Does this port need to be exposed remotely over Tor or to the local network?

If not we can remove this block, it'll still be exposed to the `bitfeed/client` container on the private internal network.

**Review by lukechilds — CHANGES_REQUESTED** [review]
Awesome stuff @mononaut, this is beautiful!

Tested and working great, just a minor thing. It works fine locally, the websocket connects via `ws://umbrel.local:8314/ws/txs`. However when trying to connect over Tor, the websocket is connecting to `wss://foo.onion/ws/txs` and failing. Do you know why it's attempting to connect over `wss` instead of `ws`?

If this is complicated to fix we don't need to block shipping this, we can disable Tor support for the intial release and re-enable it in a future release if you can resolve the issue.

File: apps/bitfeed/docker-compose.yml (hunk 1 above)
Comment (mononaut) [reply apps/bitfeed/docker-compose.yml]: Yeah it only needs to be accessible to the web container, I'll remove that block.

**Review by lukechilds — APPROVED** [review]

**Comment by lukechilds** [comment]
Great, working perfectly for me now!

LFG 🚀

**Comment by mayankchhabra** [comment]
Congrats on the launch, @mononaut! What a beautiful creation Bitfeed is! Well done.

[Here](https://github.com/getumbrel/umbrel-apps-gallery/tree/master/bitfeed) are the app store gallery images and app icon for Bitfeed. If you'd like to update them in the future, feel free to directly open a PR at [getumbrel/umbrel-apps-gallery](https://github.com/getumbrel/umbrel-apps-gallery). Cheers!

**Pull request merged**